pkg/backend/openshiftcluster.go

```diff
@@ -148,6 +148,16 @@ func (ocb *openShiftClusterBackend) handle(ctx context.Context, log *logrus.Entr
 
 		stop()
 
+		_, err = ocb.db.Billing.Patch(ctx, doc.ID, func(billingdoc *api.BillingDocument) error {
```

Comment:
(I might be wrong, but this is how I read it): If we execute delete and fail on updateAsyncOperation (cluster document at this point is `Failed`, so not dequeued), this will never be executed. And billing will be running, potentially for the non-existing cluster. I think this logic needs to change little bit. @jim-minter do you have any ideas?

Reply:
I think this is ok, billing service should be able to resolve that. 
@shrutir25 ?

Reply:
this should go in pkg/backend/openshiftcluster/delete.go

Reply:
Moving this to backend operation should be ok. We want to update the deletion time for the record after the cluster has been deleted.